### PR TITLE
Reorganize the git repository to allow direct pip install and simpler release management

### PR DESCRIPTION
**Pull request by sspickle**
Hi Folks,

Here's my attempt to reorganize the repository to make it simpler to manage releases and also permit pip to install directly from git. 

thanks,
-steve

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -29,8 +27,6 @@
           'Topic :: Multimedia :: Graphics :: 3D Rendering',
           'Topic :: Scientific/Engineering :: Visualization',
     ],
-    install_requires=['jupyter', 'vpython'],
-    #cmdclass=cmdclass('vpnotebook/data'),
-    #package_data={'vpython': ['data/*'], 'vpnotebook': ['data/kernel.json']},
+    install_requires=['jupyter', 'vpnotebook'],
```
Comment (mwcraig) [inline setup.py]: `numpy` is also currently a dependency, so I'd list it here. When that dependency is removed we can always update `install_requires`.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
 
+Run VPython in an IPython/Jupyter notebook, using the GlowScript VPython API, documented in the VPython Help at glowscript.org. VPython makes it unusually easy to create navigable real-time 3D animations. The one-line program "box()" produces a 3D cube with appropriate lighting and with the camera positioned so that the scene fills the view. It also turns on mouse interactions to zoom and rotate the camera view. The 3D scene appears in the Jupyter notebook, using the WebGL-based GlowScript 3D graphics library. This implementation of VPython was begun by John Coady in May 2014. Ruth Chabay and Bruce Sherwood are assisting in its further development. The repository for the source code is at https://github.com/BruceSherwood/vpython-jupyter.
```
Comment (mwcraig) [inline README.md]: Is this intended to replace the paragraph above (the one that begins "The IPython/Jupyter version of....")? Looks like it is, so I'd delete that version.

Comment (mwcraig) [inline README.md]: Also, I'd change "IPython/Jupyter" → "Jupyter"

File: setup.py [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
-#try:
-#    from vpnotebook import cmdclass
-#except:
-#    import pip, importlib
-#    pip.main(['install', 'vpython']); cmdclass = importlib.import_module('vpnotebook').cmdclass
+from __future__ import print_function
+try:
+    from setuptools import setup
+except:
+    print("no setuptools, using distutils")
```
Comment (mwcraig) [inline setup.py]: Might it be better to include [ez_setup.py](http://pythonhosted.org/setuptools/setuptools.html#using-setuptools-without-bundling-it) in the repo and use that to bootstrap a working `setuptools`?

**Comment by mwcraig** [comment]
One issue, which turns out to be related to whether to support `distutils` if `setuptools` is not available: right now when I create a sdist the README, CHANGES, etc are not include in the source distribution. When I try to `pip install` from the sdist I create, the install fails because `setup.py` tries to read from `README.md`, which is not present in the sdist.

~~Two~~ One option~~s~~ for fixing:

1. ~~Works only with `setuptools`: add `include_package_data=True` to the `setup` arguments and [all files under version control will be included](http://pythonhosted.org/setuptools/setuptools.html#including-data-files).~~ WORKS ONLY with subversion or CVS, not git.
2. Works with both `distutils` and `setuptools`: Create a file called `MANIFEST.in` in same directory as `setup.py` that lists files that should be included in the sdist. So our manifest might look like this:

```
include CHANGES.txt
include LICENSE.txt
include README.md
```

I'll open a PR against the branch you are working on to add `MANIFEST.in`. When you merge that PR it will automagically update your PR here.

File: setup.py (hunk 1 above)
Comment (sspickle) [inline setup.py]: I believe that dependency has been removed, or at least I think that was the intent. Yes, I just checked and it seems to be gone now.

**Comment by sspickle** [comment]
Thanks Matt for all the feedback. I think I've addressed all your comments. Let me know if I've left something out. Thx.

**Comment by sspickle** [comment]
Are we ready to merge this one? Let me know if you have reservations, otherwise I'll go ahead and merge it this afternoon. Thanks!

**Comment by BruceSherwood** [comment]
I trust you and Matt to make the decision;

**Comment by sspickle** [comment]
Huh? Bruce, did you mean to close it? I'm guessing that was an accident. I'll re-open and merge.

-steve

**Comment by sspickle** [comment]
OK! I just re-opened and merged it.

You’ll want to ‘git pull’ to get the new format on your local system before checking in new code.

thanks!
-steve

**Comment by BruceSherwood** [comment]
Apologies. I hit the wrong button.

Bruce

File: setup.py (hunk 1 above)
Comment (sspickle) [inline setup.py]: Oops. It's gone from vpython.py, but it's still in rate_control.py ;-(. Sorry, I missed that.

Bruce, should I work on removing numpy from rate_control? Is that something you intend to have done by someone?

thx.
-steve

Comment (BruceSherwood) [inline setup.py]: Oy! It's good that you caught that. Let me see whether I can deal with
this, as I'd like to put this all together today.

Bruce

On Mon, Jan 25, 2016 at 6:52 AM, Steve Spicklemire <notifications@github.com

**Pull request merged**